### PR TITLE
deifentification-core

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: deidentification-core/src/main/java/gov/va/api/health/deidentification/core/SyntheticData.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package gov.va.api.health.deidentification.core;
+
+/**
+ * The SyntheticData interface defines the set of FHIR PII fields that must be deidentified.
+ * Implementations will define the details of deidentification. For example, class path resource
+ * based implementation exists in ResourceBasedSyntheticData.java
+ */
+public interface SyntheticData {
+
+  String synthesizeDate(String rawBirthdate);
```
Comment (jhulbert-va) [inline deidentification-core/src/main/java/gov/va/api/health/deidentification/core/SyntheticData.java:10]: Nitpick: This should just be `rawDate`. The interface shouldn't know if its a birthdate or not.

**Review by jhulbert-va — CHANGES_REQUESTED** [review]

File: deidentification-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    <version>4.0.10-SNAPSHOT</version>
+  </parent>
+  <artifactId>deidentification-core</artifactId>
+  <version>4.0.10-SNAPSHOT</version>
+  <packaging>jar</packaging>
+  <properties>
+    <service-auto-config.version>7.0.10</service-auto-config.version>
+    <java-uuid-generator.version>3.1.5</java-uuid-generator.version>
+  </properties>
+  <dependencies>
```
[the remark this answers is not in the document]
Comment (ilaflamme-va) [reply deidentification-core/pom.xml:16]: recommendation?

File: deidentification-core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>fhir-resources-parent</artifactId>
+    <groupId>gov.va.api.health</groupId>
+    <version>4.0.17-SNAPSHOT</version>
+  </parent>
+  <artifactId>deidentification-core</artifactId>
+  <version>4.0.15-SNAPSHOT</version>
```
Comment (jhulbert-va) [inline deidentification-core/pom.xml:10]: Rev this boi

File: deidentification-core/pom.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  </properties>
+  <dependencies>
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <version>1.2.3</version>
```
Comment (jhulbert-va) [inline deidentification-core/pom.xml:25]: Can we pull the rest of these versions into properties?

**Review by jhulbert-va — CHANGES_REQUESTED** [review]

File: deidentification-core/src/main/java/gov/va/api/health/deidentification/core/ResourceBasedSyntheticData.java [hunk 5]
```diff
@@ -32,7 +31,7 @@
    */
   @Override
   public String synthesizeDate(String rawDate) {
-    if (StringUtils.isBlank(rawDate)) {
+    if (rawDate == null || rawDate.isEmpty()) {
```
Comment (jhulbert-va) [inline deidentification-core/src/main/java/gov/va/api/health/deidentification/core/ResourceBasedSyntheticData.java:34]: Should this be `isBlank()`?

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**